### PR TITLE
PLAT-129759: TabGroup: Added JSDoc and fixed tab border-left bug

### PR DESCRIPTION
**Pull request by daniel-stoian-lgp**
### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [x] A CHANGELOG entry is included
* [x] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
Issues: 
agate/TabGroup needed an architecture sync
agate/TabGroup was missing JSDoc for its props
agate/Tabgroup was displaying border-left for tabs even for vertical orientation

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)
- analysis and local test did not show the need to change architecure. Check user story comment
- added JSDoc for props
- kept styling for tab `border-left` only for horizontal `orientation`
- added styling for tab `border-bottom` for horizontal `vertical`

### Links
[//]: # (Related issues, references)
PLAT-129759

### Comments

Enact-DCO-1.0-Signed-off-by: Daniel Stoian daniel.stoian@lgepartner.com

### CODE REVIEW

File: TabGroup/TabGroup.js [hunk 1]
```diff
@@ -30,11 +30,60 @@ const TabBase = kind({
 	name: 'Tab',
 
 	propTypes: {
+		/**
+		 * Customizes the component by mapping the supplied collection of CSS class names to the
+		 * corresponding internal elements and states of this component.
+		 *
+		 * The following classes are supported:
+		 *
+		 * * `tab` - The root class name
```
Comment (adrian-cocoara-lgp) [inline TabGroup/TabGroup.js:39]: Remove one *

File: TabGroup/TabGroup.js [hunk 2]
```diff
@@ -30,11 +30,60 @@ const TabBase = kind({
...
+		 *
+		 * @type {Function}
+		 * @public
+		 */
 		onClick: PropTypes.func,
+
+		/**
+		 * Orientation of the tab.
+		 *
+		 * * Values: `'horizontal'`, `'vertical'`
```
Comment (adrian-cocoara-lgp) [inline TabGroup/TabGroup.js:73]: Remove one *

File: TabGroup/TabGroup.js [hunk 3]
```diff
@@ -108,12 +157,68 @@ const TabGroupBase = kind({
...
+		 *
+		 * @type {Object}
+		 * @public
+		 */
 		css: PropTypes.object,
+
+		/**
+		 * Orientation of the tabs.
+		 *
+		 * * Values: `'horizontal'`, `'vertical'`
```
Comment (adrian-cocoara-lgp) [inline TabGroup/TabGroup.js:209]: Remove one *

File: TabGroup/TabGroup.js [hunk 4]
```diff
@@ -108,12 +157,68 @@ const TabGroupBase = kind({
...
+		 */
 		beforeTabs: PropTypes.node,
+
+		/**
+		 * Customizes the component by mapping the supplied collection of CSS class names to the
+		 * corresponding internal elements and states of this component.
+		 *
+		 * The following classes are supported:
+		 *
+		 * * `tabBar` - The root class name
```
Comment (adrian-cocoara-lgp) [inline TabGroup/TabGroup.js:199]: Remove one *

**Review by adrian-cocoara-lgp — APPROVED** [review]
LGTM

File: TabGroup/TabGroup.js [hunk 5]
```diff
@@ -108,12 +165,68 @@ const TabGroupBase = kind({
 	name: 'TabGroup',
 
 	propTypes: /** @lends agate/TabGroup.TabGroupBase.prototype */ {
+		/**
+		 * The position of the TabGroup related to the tab contents.
+		 *
+		 * @type {String}
+		 * @required
+		 */
```
Comment (MikyungKim) [inline TabGroup/TabGroup.js:173]:
```suggestion
		 * @public
		 */
```

File: TabGroup/TabGroup.js [hunk 6]
```diff
@@ -108,12 +165,68 @@ const TabGroupBase = kind({
...
+		 * The provided list of tabs.
+		 *
+		 * @type {Array}
+		 * @required
+		 * @public
+		 */
 		tabs: PropTypes.array.isRequired,
+
+		/**
+		 * Nodes to be inserted before the tabs.
```
Comment (MikyungKim) [inline TabGroup/TabGroup.js:186]:
```suggestion
		 * Nodes to be inserted after the tabs.
```

File: TabGroup/TabGroup.js [hunk 7]
```diff
@@ -108,12 +165,68 @@ const TabGroupBase = kind({
...
+		 *
+		 * @type {Node}
+		 * @public
+		 */
 		afterTabs: PropTypes.node,
+
+		/**
+		 * Nodes to be inserted after the tabs.
```
Comment (MikyungKim) [inline TabGroup/TabGroup.js:194]:
```suggestion
		 * Nodes to be inserted before the tabs.
```

File: TabGroup/TabGroup.module.less [hunk 8]
```diff
@@ -20,7 +20,6 @@
 				color: @agate-panels-tab-color;
 				// background-color: @agate-panels-tab-bg-color;
 				// background-image: @agate-panels-tab-bg-image;
-				border-left: @agate-panels-tab-separator-border-width solid @agate-panels-tab-separator-border-color;
```
Comment (MikyungKim) [inline TabGroup/TabGroup.module.less:23]: For `vertical`, should we add `border-bottom`? WDYT?

File: TabGroup/TabGroup.js [hunk 9]
```diff
@@ -26,15 +26,72 @@ import ToggleButton from '../ToggleButton';
...
+		 * @type {Object}
+		 * @public
+		 */
 		css: PropTypes.object,
+
+		/**
+		 * The icon displayed on the tab.
+		 *
+		 * @type {String}
+		 * @public
```
Comment (MikyungKim) [inline TabGroup/TabGroup.js:58]:
```suggestion
		 * @default 'star'
		 * @public
```

File: TabGroup/TabGroup.module.less (hunk 8 above)
Comment (daniel-stoian-lgp) [reply TabGroup/TabGroup.module.less:23]: I agree

File: CHANGELOG.md [hunk 10]
```diff
@@ -32,6 +32,7 @@ The following is a curated list of changes in the Enact agate module, newest cha
 - `agate/LabeledIconButton` to display label text on multiple lines (removed marquee)
 - `agate/LabeledIconButton` `max-width` to display huge sized icon correctly
 - `agate/MediaPlayer` previous button functionality to play media from the beginning after being paused
+- `agate/TabGroup` tab `border-left` to not be applied for vertical orientation
```
Comment (MikyungKim) [inline CHANGELOG.md:35]:
```suggestion
- `agate/TabGroup` tab `border-bottom` to be applied for vertical orientation
```

File: TabGroup/TabGroup.js [hunk 11]
```diff
@@ -26,15 +26,73 @@ import ToggleButton from '../ToggleButton';
...
+		 */
 		onClick: PropTypes.func,
+
+		/**
+		 * Orientation of the tab.
+		 *
+		 * * Values: `'horizontal'`, `'vertical'`
+		 *
+		 * @type {String}
```
Comment (MikyungKim) [inline TabGroup/TabGroup.js:84]:
```suggestion
		 * @type {('horizontal'|'vertical')}
```

File: TabGroup/TabGroup.js [hunk 12]
```diff
@@ -26,15 +26,73 @@ import ToggleButton from '../ToggleButton';
...
+		 * @public
+		 */
 		orientation: PropTypes.string,
```
Comment (MikyungKim) [inline TabGroup/TabGroup.js:87]:
```suggestion
		orientation: PropTypes.oneOf(['horizontal', 'vertical']),,
```

File: TabGroup/TabGroup.js [hunk 13]
```diff
@@ -108,12 +166,69 @@ const TabGroupBase = kind({
...
+		 *
+		 * @type {Object}
+		 * @public
+		 */
 		css: PropTypes.object,
+
+		/**
+		 * Orientation of the tabs.
+		 *
+		 * * Values: `'horizontal'`, `'vertical'`
```
Comment (MikyungKim) [inline TabGroup/TabGroup.js:219]: Here, please do the same thing above.

**Review by MikyungKim — APPROVED** [review]
LGTM!

File: TabGroup/TabGroup.js [hunk 14]
```diff
@@ -26,15 +26,73 @@ import ToggleButton from '../ToggleButton';
 
 import componentCss from './TabGroup.module.less';
 
+/**
+ * A Tab component.
+ *
+ * @class TabBase
+ * @memberof agate/TabGroup
+ * @ui
+ * @public
```
Comment (jeonghee27) [inline TabGroup/TabGroup.js:35]: Because the TabBase is not exported. it should be private. I'll Fix this part.

**Pull request merged**